src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,197 @@
+            Paths.get("src/main/java/com/puppycrawl/tools/checkstyle/checks");
+
+    public static final String NODE_NAME = "a";
+
+    public static final String SUFFIX_JAVA = ".java";
+
+    public static final String SUFFIX_CHECK_JAVA = "Check.java";
+
+    public static final String MISC = "misc";
+
+    public static final String ANNOTATION = "annotation";
+
+    public static final String COMMENTS_INDENTATION = "CommentsIndentation";
+
+    public static final String INDENTATION = "Indentation";
+
+    public static final String SUPPRESS_WARNINGS_HOLDER = "SuppressWarningsHolder";
+
+    public static final Path AVAILABLE_CHECKS_PATH = Paths.get("src/xdocs/checks.xml");
+
+    private static Map<String, String[]> getXdocsMap() {
+        final File checks = new File(PACKAGE_NAME_PATH.toString());
+        final String[] checksSubDirNames = checks.list((rootDir, pathname) -> {
+            return !pathname.endsWith(SUFFIX_JAVA);
+        });
+        final Map<String, String[]> checksNamesMap = new HashMap<>();
+        if (checksSubDirNames != null) {
+            final FilenameFilter filter = (rootDir, pathname) -> {
+                return pathname.endsWith(SUFFIX_CHECK_JAVA);
+            };
+            for (String dirName : checksSubDirNames) {
+                final File dirFile = new File(PACKAGE_NAME_PATH + "/" + dirName);
+                final String[] fileNames = dirFile.list(filter);
+                checksNamesMap.put(dirName, fileNames);
+            }
+            final String[] miscNames = checks.list(filter);
+            checksNamesMap.put(MISC, miscNames);
+        }
+        return checksNamesMap;
+    }
+
+    @Test
+    public void testXdocsUrl() throws Exception {
+        final SAXParserFactory parserFactory = SAXParserFactory.newInstance();
+        final SAXParser parser = parserFactory.newSAXParser();
+        final CheckTest checkHandler = new CheckTest(NODE_NAME);
+        final List<List<String>> list;
+        try (InputStream input = Files.newInputStream(AVAILABLE_CHECKS_PATH)) {
+            parser.parse(input, checkHandler);
+            list = checkHandler.list;
+        }
+        final Map<String, String[]> checksNamesMap = getXdocsMap();
+        for (List<String> sub : list) {
+            final String moduleName = sub.get(0);
+            final String checkName1 = sub.get(1);
+            final String checkName2 = sub.get(2);
+            final String errorMessage =
+                    "URL in" + AVAILABLE_CHECKS_PATH + " is inconsistent: "
+                            + moduleName + " " + checkName1 + " " + checkName2;
+            assertEquals(checkName1, checkName2, errorMessage);
+            if (checkName1.equals(COMMENTS_INDENTATION) || checkName1.equals(INDENTATION)) {
+                assertEquals(MISC, moduleName, errorMessage);
+            }
+            else if (checkName1.equals(SUPPRESS_WARNINGS_HOLDER)) {
+                assertEquals(ANNOTATION, moduleName, errorMessage);
```

Comment:
`assertEquals`
For new unit tests google Truth assertWithMessage should be used

Reply:
done